src/index.ts

```diff
@@ -1,15 +1,15 @@
-export default function doublet<Arg, R>(cb: Callback<Arg, R>, ...args: Array<Arg>): MaybeAsyncResult<R> {
+export default function doublet<TCallback extends Callback>(cb: TCallback, ...args: Parameters<TCallback>): MaybeAsyncResult<ReturnType<TCallback>> {
   try {
-    const result = cb(...args);
+    const result = cb(...args as Array<unknown>);
```

Comment:
This little cast is somehow needed since `Parameters<T>` returns a tuple but that is not very obvious when looking at the types.